File: app/src/main/java/com/lasercyber/lws/ui/bean/http/ApiResult.java
package com.lasercyber.lws.ui.bean.http;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import lombok.Data;
import lombok.experimental.Accessors;

/**
 * Standard Worker JSON envelope (same shape as TS {@code ApiResult<T>}).
 * <p>
 * Success is determined only by {@code success === true}. Do not infer success from {@code code}.
 */
@Data
@Accessors(chain = true)
public class ApiResult implements Serializable {
    @SerializedName("success")
    private Boolean success;
    @SerializedName("code")
    private Integer code;
    @SerializedName("message")
    private String message;
    @SerializedName("data")
    private Object data;

    public boolean isSuccess() {
        return Boolean.TRUE.equals(success);
    }
}

File: app/lws_hmi/lib/platform/local_http/api_result.dart
import 'dart:convert';

/// LAN JSON envelope (`ApiResult`) for `:5580`.
final class ApiResult {
  const ApiResult({
    required this.success,
    required this.code,
    required this.message,
    this.data,
  });

  final bool success;
  final int code;
  final String message;
  final Object? data;

  static ApiResult ok({Object? data, String message = 'ok'}) =>
      ApiResult(success: true, code: 0, message: message, data: data);

  static ApiResult fail(
    String message, {
    int code = 1,
    Object? data,
  }) =>
      ApiResult(success: false, code: code, message: message, data: data);

  Map<String, Object?> toJson() => {
        'success': success,
        'code': code,
        'message': message,
        'data': data,
      };

  String encode() => jsonEncode(toJson());
}
